stl/inc/algorithm

```diff
@@ -2900,6 +2900,89 @@ _FwdIt3 transform(_ExPo&& _Exec, const _FwdIt1 _First1, const _FwdIt1 _Last1, co
+
+    // ALIAS TEMPLATE unary_transform_result
+    template <class _In, class _Out>
+    using unary_transform_result = in_out_result<_In, _Out>;
+
+    // ALIAS TEMPLATE binary_transform_result
+    template <class _In1, class _In2, class _Out>
+    using binary_transform_result = in_in_out_result<_In1, _In2, _Out>;
+
+    // VARIABLE ranges::transform
+    class _Transform_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, sentinel_for<_It> _Se, weakly_incrementable _Out,
+            copy_constructible _Fn, class _Pj = identity>
+            requires indirectly_writable<_Out, indirect_result_t<_Fn&, projected<_It, _Pj>>>
+        _NODISCARD constexpr unary_transform_result<_It, _Out> operator()(_It _First, _Se _Last, _Out _Result,
+            _Fn _Func, _Pj _Proj = {}) const {
+            // transform [_First, _Last) with _Func
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            auto _UResult     = _Get_unwrapped(_STD move(_Result));
+            for (; _UFirst != _ULast; ++_UFirst, (void) ++_UResult) {
+                *_UResult = _STD invoke(_Func, _STD invoke(_Proj, *_UFirst));
+            }
+
+            _Seek_wrapped(_First, _STD move(_UFirst));
+            _Seek_wrapped(_Result, _STD move(_UResult));
+            return {_STD move(_First), _STD move(_Result)};
+        }
+
+        template <input_range _Rng, weakly_incrementable _Out, copy_constructible _Fn, class _Pj = identity>
+            requires indirectly_writable<_Out, indirect_result_t<_Fn&, projected<iterator_t<_Rng>, _Pj>>>
+        _NODISCARD constexpr unary_transform_result<borrowed_iterator_t<_Rng>, _Out>
+            operator()(_Rng&& _Range, _Out _Result, _Fn _Func, _Pj _Proj = {}) const {
+            return (*this)(_RANGES begin(_Range), _RANGES end(_Range), _STD move(_Result), _Pass_fn(_Func),
+                _Pass_fn(_Proj));
+        }
+
+        template <input_iterator _It1, sentinel_for<_It1> _Se1, input_iterator _It2, sentinel_for<_It2> _Se2,
+            weakly_incrementable _Out, copy_constructible _Fn, class _Pj1 = identity, class _Pj2 = identity>
+            requires indirectly_writable<_Out, indirect_result_t<_Fn&, projected<_It1, _Pj1>, projected<_It2, _Pj2>>>
+        _NODISCARD constexpr binary_transform_result<_It1, _It2, _Out> operator()(_It1 _First1, _Se1 _Last1,
+            _It2 _First2, _Se2 _Last2, _Out _Result, _Fn _Func, _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            // transform [_First1, _Last1) and [_First2, ...) with _Func
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+            auto _UFirst1      = _Get_unwrapped(_STD move(_First1));
+            const auto _ULast1 = _Get_unwrapped(_STD move(_Last1));

```

Comment:
I think this should become `_Get_final_iterator_unwrapped` once it is merged

Reply:
We don't want to potentially iterate over the entire sequence to find the last iterator value in `_Get_final_iterator_unwrapped`. Especially when the loop below is about to iterate over the entire sequence and find the final iterator value in the process.

Reply:
That makes sense